lib/pluginmanager/bundler/logstash_uninstall.rb

```diff
@@ -45,43 +45,47 @@ def dependants_gems(gem_name)
         .collect(&:name).sort.uniq
     end
 
-    def uninstall!(gem_name)
-      unfreeze_gemfile do
-        dependencies_from = dependants_gems(gem_name)
-
-        if dependencies_from.size > 0
-          display_cant_remove_message(gem_name, dependencies_from)
-          false
-        else
-          remove_gem(gem_name)
-          true
+    def uninstall!(gem_names)
+      gem_names = [gem_names] if gem_names.kind_of?(String)
+
+      mutate_gemfile do |gemfile|
+        builder = Dsl.new
+        # create a copy of our gemfile and optimistically remove all the gems
+        gem_names.each { |gem_name| gemfile.remove(gem_name) }
+        builder.eval_gemfile(::File.join(::File.dirname(gemfile_path), "gemfile to changes"), gemfile.generate)
+
+        # build a definition, providing an intentionally-empty "unlock" mapping
+        # to ensure that all gem versions remain locked
+        definition = builder.to_definition(lockfile_path, {})
+
+        # seek for conflicts, specifically still-existing gems that
+        # have runtime dependencies on the ones we have optimistically removed
+        conflicts = gem_names.each_with_object({}) do |gem_to_remove, memo|
```

Comment:
For this section we are searching all specs for each of the gems/plugins in the list. In the spirit of scaling this to multiple removals at a time, i think it would make more sense to run the the entire spec list once and look for any instance of the list of gems to be removed. 

Current: 
- Look through ALL gems to see what depends on it
- Repeat this search again from scratch for each gem we want to remove

Suggestion:
- Look through ALL gems just once
- During that single look, check for any dependencies on ANY of our removal gems